src/app/archive/create_schema_new.sql

```diff
@@ -0,0 +1,73 @@
+CREATE TABLE state_hashes
```

Comment:
You can just put the state hash directly on the `blocks` table and add an index for it (`CREATE INDEX idx_block_state_hash ON blocks(state_hash)`). It's not a big deal to have this separate though if there is a reason for it, but having it this way would add some mild indirection to finding a block associated with some state hash coming from a firehose event.